### PR TITLE
LPS-197642 Fixing the behavior for the selected account remain selected after change page

### PR DESCRIPTION
**Pull request by ytuik**
https://liferay.atlassian.net/browse/LPS-197642

### CODE REVIEW

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/AccountSelection.tsx [hunk 1]
```diff
@@ -29,16 +33,27 @@ const AccountSelection = ({onSelectAccount}: AccountSelectionProps) => {
...
-				(display, roleBrief) => {
-					if (enabledAccountRoles.includes(roleBrief.name)) {
-						return true;
-					}
-
-					return display;
-				},
-				false
-			);
+			let displayAccount: boolean = false;
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/AccountSelection.tsx:36]:
```suggestion
			let displayAccount = false;
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/AccountSelection.tsx [hunk 2]
```diff
@@ -29,16 +33,27 @@ const AccountSelection = ({onSelectAccount}: AccountSelectionProps) => {
...
+			if (!accountBrief.roleBriefs.length) {
+				const accountInfo: Account = await getAccountInfo({
+					accountId: Number(accountBrief.id),
+				});
+				if (accountInfo.type === 'person') {
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/AccountSelection.tsx:41]:
```suggestion

				if (accountInfo.type === 'person') {
```

Comment (kevenleone) [reply workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/AccountSelection.tsx:41]: Space

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/AccountSelection.tsx [hunk 3]
```diff
@@ -29,16 +33,27 @@ const AccountSelection = ({onSelectAccount}: AccountSelectionProps) => {
...
+				displayAccount = accountBrief.roleBriefs.reduce(
+					(display, roleBrief) => {
+						if (enabledAccountRoles.includes(roleBrief.name)) {
+							return true;
+						}
+
+						return display;
+					},
+					false
+				);
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/AccountSelection.tsx:55]: I think the result is the same, can you test?

```suggestion
				displayAccount = accountBrief.roleBriefs.some(
					(roleBrief) => enabledAccountRoles.includes(roleBrief.name))
				);
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/AccountSelection.tsx [hunk 4]
```diff
@@ -47,6 +62,12 @@ const AccountSelection = ({onSelectAccount}: AccountSelectionProps) => {
 
 				radioAccountList.push({
 					imageURL: accountInfo.logoURL,
+					selected:
+						selectedAccount &&
+						selectedAccount.externalReferenceCode ===
+							accountInfo.externalReferenceCode
+							? true
+							: false,
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/AccountSelection.tsx:70]:
```suggestion
					selected: 
					    selectedAccount?.externalReferenceCode === accountInfo.externalReferenceCode
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/AccountSelection.tsx [hunk 5]
```diff
@@ -58,8 +79,26 @@ const AccountSelection = ({onSelectAccount}: AccountSelectionProps) => {
...
+				if (index === radioOption.index) {
+					account.selected = true;
+
+					return account;
+				}
+				account.selected = false;
+
+				return account;
+			})
+		);
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/AccountSelection.tsx:99]:
```suggestion
		setAccounts((prevAccounts) =>
			prevAccounts.map((account, index) => ({
				...account,
				selected: index === radioOption.index
			}))
		);
```

**Review by kevenleone — CHANGES_REQUESTED** [review]
Bruno, code looks good 🎉 

I left few suggestions for you

**Comment by ytuik** [comment]
ci:test:sf

**Review by kevenleone — APPROVED** [review]
LGTM 🔥 

Good job @ytuik !

**Comment by kevenleone** [comment]
ci:forward

**Pull request closed without merging**